search-sentiments-map/public/scripts/modal-script.js

```diff
@@ -105,50 +106,72 @@ async function displayTopResultsForCurrentTrend(countryCode) {
-              snippetArray.forEach((word) => {
-                // Replace quotations or other symbols that may be attached to 
-                // the word.  
-                let checkWordInArray = word.replace(/[^a-z0-9+]+/gi, '');
-                if (results[i].score > 0 && sentimentWordsResult.positive
-                    .includes(checkWordInArray)) {
-                  resultElement.innerHTML += '<span style=\'color: ' + 
-                      POSITIVE_COLOR + ';\'>' + word + ' </span>';
-                } else if (results[i].score < 0 && sentimentWordsResult.negative
-                    .includes(checkWordInArray)) {
-                  resultElement.innerHTML += '<span style=\'color: ' + 
-                      NEGATIVE_COLOR + ';\'>' + word + ' </span>';
-                } else {
-                  resultElement.innerHTML += word + ' ';
-                }
-              });
-              resultElement.innerHTML += '<br><i>Sentiment Score: ' + 
-                  results[i].score.toFixed(1) + '</i><br>';
-            } else {
-              resultElement.innerHTML += results[i].snippet + '<br>' + 
-                  '<i>Sentiment Score: ' + results[i].score.toFixed(1) + 
-                  '</i><br>';
+
+            // Check that positive and negative words are detected by the Node.js 
+            // sentiment API.
+            if (sentimentWordsResult != null) {
+              let positiveWords = sentimentWordsResult.positive;
+              let negativeWords = sentimentWordsResult.negative;
+              if (results[i].score > 0 && positiveWords.length !== 0) {
+                highlightWords(
+                  snippet, new Set(positiveWords), true, resultElement);
+              } else if (results[i].score < 0 && negativeWords.length !== 0) {
+                highlightWords(
+                  snippet, new Set(negativeWords), false, resultElement);
+              } else {
+                resultElement.innerHTML += snippet;
+              }
             }
+            resultElement.innerHTML += '<br><i>Sentiment Score: ' + 
+                  results[i].score.toFixed(1) + '</i><br>';
           });
     }
     resultElement.innerHTML += '<i>Last updated on ' + date + '</i>';
   }
 }
 
+/** 
+ * Highlights the positive words green if the sign of the sentiment score 
+ * is positive and negative words red if the sign is negative.
+ * @param {string} snippet Snippet to display in the search results tab.
+ * @param {Set} wordsToHighlight Words to highlight a different color.
+ * @param {boolean} scoreIsPositive Boolean for whether or not the sentiment 
+ *     score is positive.
+ * @param {Object} resultElement Tab element to update with the search result 
+ *     snippet.
+ */
+function highlightWords(
+    snippet, wordsToHighlight, scoreIsPositive, resultElement) {
+  let startDisplayingSnippetIndex = 0;
+  wordsToHighlight.forEach((word) => {
+    let wordIndex = snippet.toLowerCase().indexOf(word);
```

Comment:
Add a comment explaining why you only need to transform the snippet to lower case, and not the word too.